backend/csp/calculateProbabilities.ts

```diff
@@ -0,0 +1,114 @@
+import { MatchBoxSchema } from "../db/schemas/matchBox.ts";
+import { MatchNightSchema } from "../db/schemas/matchNight.ts";
+import { AYO } from "./AYOcsp.ts";
+
+// deno-lint-ignore no-explicit-any
+function json2array(json: any) { // skipcq: JS-0323
+  // deno-lint-ignore no-explicit-any
+  const result: any[] = []; // skipcq: JS-0323
+  const keys = Object.keys(json);
+  keys.forEach(function (key) {
+    result.push(json[key]);
+  });
+  return result;
+}
+
+export async function getCurrenProbabilities(
+  currentSeason: number,
+  currentEpisode: number,
+) {
+  const man: Array<string> = [];
+  const woman: Array<string> = [];
+  let person11 = "";
+  const matchbox: Array<MatchBoxSchema> = [];
+  const matchnight: Array<MatchNightSchema> = [];
+
+  // get data from db
+  const matchboxes = await fetch(
+    "http://dayo-project.herokuapp.com/api/v1/matchboxes/",
+  )
+    .then((res) => res.json());
+  const matchnights = await fetch(
+    "http://dayo-project.herokuapp.com/api/v1/matchnights/",
+  )
+    .then((res) => res.json());
+  const participants = await fetch(
+    "http://dayo-project.herokuapp.com/api/v1/participants",
+  )
+    .then((res) => res.json());
```

Comment:
Maybe use direct database interaction, such as in the commands. The backend from our web application will be offline most of the time and responses might be slower. Also, if using direct database interaction, the bot will work on its own.

Reply:
This will also slim down the code base because, no json stuff is required